src/uploaders/lib/EndpointUrl.ts

```diff
@@ -0,0 +1,8 @@
+export const DEFAULT_UPLOAD_ORIGIN = 'https://upload.bugsnag.com'
+
+export function buildEndpointUrl (origin: string, path: string): string {
+  const url = new URL(origin)
+  // if more than a full origin was given, use the provided URL as-is
+  if (url.pathname !== '/') return url.toString()
+  return new URL(path, origin).toString()
+}
```

Comment:
Not sure this is better, but it's possible to avoid constructing two `URL`s here

```suggestion
export function buildEndpointUrl (origin: string, path: string): string {
  const url = new URL(origin)
  
  // if no path was given use the default
  if (url.pathname === '/') {
    url.pathname = path
  }

  return url.toString()
}
```